WD-Firmware/comrv/comrv.c

```diff
@@ -18,34 +18,765 @@
+         stComrvCB.stOverlayCache[ucEntryIndex].ucNextIndex           = ucIndex;
+         stComrvCB.stOverlayCache[ucEntryIndex].ucPrevIndex           = D_COMRV_LRU_FIRST_ITEM;
+         /* mark the group ID so that it won't pop in the next search */
+         stComrvCB.stOverlayCache[ucEntryIndex].unToken.uiValue       = D_COMRV_ENTRY_TOKEN_INIT_VALUE;
+         stComrvCB.stOverlayCache[ucEntryIndex].unProperties.ucValue &= D_COMRV_ENTRY_PROPERTIES_RESET_MASK;
+         stComrvCB.ucLruIndex = ucIndex;
+#elif defined(D_COMRV_EVICTION_LFU)
+#elif defined(D_COMRV_EVICTION_MIX_LRU_LFU)
+#endif /* D_COMRV_EVICTION_LRU */
+      }
+      // TODO: protect the soon loaded ram
+      /* it is safe now to get new requests */
+      M_COMRV_EXIT_CRITICAL_SECTION();
+      /* the group size in bytes */
+      usOverlayGroupSize <<= 9;
+      /* now we can load the overlay group */
+      pAddress = comrvLoadOvlayGroupHook(M_COMRV_GET_GROUP_OFFSET_IN_BYTES(stComrvCB.stOverlayCache[ucIndex].unToken),
+            stComrvCB.stOverlayCache[ucIndex].pFixedEntryAddress, usOverlayGroupSize);
+      /* if group wasn't loaded */
+      if (pAddress == NULL)
+      {
+         comrvNotificationHook(D_COMRV_LOAD_ERR, unToken.uiValue);
+      }
+
+#ifdef D_COMRV_CRC
+      /* calculate crc */
+      uiCrc = comrvCrcCalcHook(pAddress, usOverlayGroupSize-sizeof(u32_t));
+      if (uiCrc != *((u08_t*)pAddress + (usOverlayGroupSize-sizeof(u32_t))))
+      {
+         comrvNotificationHook(D_COMRV_CRC_CHECK_ERR, unToken.ucValue);
+      }
+#endif /* D_COMRV_CRC */
+
+#ifdef D_COMRV_FW_INSTRUMENTATION
+      /* update for FW profiling loaded the function */
+      uiProfilingIndication |= D_COMRV_PROFILING_LOAD_VAL;
+#endif /* D_COMRV_FW_INSTRUMENTATION */
+   }
+   else
+   {
+      usOverlayGroupSize <<= 9;
+   }
+
+   /* get actual function/data offset */
+   usOffset = M_COMRV_GET_TOKEN_OFFSET_IN_BYTES(unToken);
+
+   /* are we returning to an overlay function */
+   if (ucIsInvoke == D_COMRV_RET_CALLER_BIT_0)
+   {
+      /* calculate the function return offset; at this point
+         pComrvStackFrame->uiCalleeToken holds the return address */
+#ifdef D_COMRV_MULTI_GROUP_SUPPORT
+      /* In multi group token we need to get the offset from previously loaded token */
+      if (pComrvStackFrame->usCalleeMultiGroupTableEntry)
+      {
+         /* we now are at the point of loading a multi-group toke so we need to take the
+            previous token for which the return address refers to */
+         unToken = pOverlayMultiGroupTokensTable[pComrvStackFrame->usCalleeMultiGroupTableEntry];
+         /* get the offset */
+         uiTemp = M_COMRV_GET_TOKEN_OFFSET_IN_BYTES(unToken);
+         /* get the token group size */
+         usOverlayGroupSize = M_COMRV_GET_OVL_GROUP_SIZE_IN_BYTES(unToken);
+         /* calculate the actual return offset */
+         usOffset += ((u32_t)(pComrvStackFrame->uiCalleeToken) - uiTemp) & (--usOverlayGroupSize);
```

Comment:
clear the value of this field usCalleeMultiGroupTableEntry at this point

Reply:
done in 64dd0f8